swift/ql/test/query-tests/Security/CWE-825/test.ql

```diff
@@ -0,0 +1,61 @@
+import swift
+import codeql.swift.dataflow.DataFlow
+import Flow::PathGraph
+
+module Conf implements DataFlow::StateConfigSig {
+  class FlowState = Callable;
+
+  predicate isSource(DataFlow::Node node, FlowState state) {
+    // parameter of closure or function passed to withUnsafeBytes
+    exists(
+      CallExpr call |
+      call.getStaticTarget().hasName("withUnsafeBytes(_:)") and
+      state = node.(DataFlow::ParameterNode).getDeclaringFunction().getUnderlyingCallable() and
+      (
+        // if the declaring callable is a closure expr
+        state = call.getArgument(0).getExpr()
+        or
+        state.(Function).getAnAccess() = call.getArgument(0).getExpr()
+      )
+    )
+  }
+
+  predicate isSink(DataFlow::Node node, FlowState state) {
+    node.(DataFlow::InoutReturnNode).getParameter().getDeclaringFunction() = state
```

Comment:
What does this case do?

Reply:
This handles the case where variables are captured and then modified in the closure - they're treated as implicit inout parameters.

Reply:
Ah, I can see that's really helpful in this query!